hera_cal/abscal_funcs.py

```diff
@@ -1059,201 +1213,124 @@ def __eq__(self, B2):
         if np.isclose(self.len, B2.len, atol=tol):
             # check x, y, z
             equiv = bool(reduce(operator.mul, map(lambda x: np.isclose(*x, atol=tol), zip(self.bl, B2.bl))))
+            dot = np.dot(self.unit, B2.unit)
             if equiv:
                 return True
             # check conjugation
-            elif np.isclose(np.arccos(np.dot(self.unit, B2.unit)), np.pi, atol=tol/self.len):
+            elif np.isclose(np.arccos(dot), np.pi, atol=tol/self.len) or (dot < -1.0):
                 return 'conjugated'
             # else return False
             else:
                 return False
         else:
             return False
 
-def smooth_data(Xtrain, ytrain, Xpred, kind='gp', n_restart=3, degree=1, ls=10.0, return_model=False):
+def match_red_baselines(model, model_antpos, data, data_antpos, tol=1.0, verbose=True):
     """
-
+    match model baseline keys to data baseline keys based on positional redundancy
 
     Parameters:
     -----------
-    Xtrain : ndarray containing x-values for regression
-        type=ndarray, dtype=float, shape=(Ndata, Ndimensions)
-  
-    ytrain : 1D np.array containing parameter y-values across time and/or frequency (single pol)
-        type=1D-ndarray, dtype=float, shape=(Ndata, Nfeatures)
-  
-    Xpred : ndarray containing x-values for prediction
-        type=ndarray, dtype=float, shape=(Ndata, Ndimensions)
-
-    kind : kind of smoothing to perform, type=str, opts=['linear','const']
-        'poly' : fit a Nth order polynomial across frequency, with order set by "degree"
-        'gp' : fit a gaussian process across frequency
-        'boxcar' : convolve ytrain with a boxcar window
-    """
-    # get number of dimensions
-    ndim = Xtrain.shape[1]
-
-    if kind == 'poly':
-        # robust linear regression via sklearn
-        model = make_pipeline(PolynomialFeatures(degree), linear_model.RANSACRegressor())
-        model.fit(Xtrain, ytrain)
-
-    elif kind == 'gp':
-        # construct GP kernel
-        ls = np.array([1e-1 for i in range(ndim)])
-        kernel = 1.0**2 * gaussian_process.kernels.RBF(ls, np.array([1e-2, 1e1])) + \
-                 gaussian_process.kernels.WhiteKernel(1e-4, (1e-8, 1e1))
-        model = gaussian_process.GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=n_restart)
-        # fit GP
-        model.fit(Xtrain, ytrain)
-
-    ypred = model.Predict(X)
+    data : type=dictionary, data dictionary holding complex visibilities.
```

Comment:
Parameters are out of order (relative to the function definition)

Reply:
fixed